### PR TITLE
Display celsius on chart and in statistics container

### PR DESCRIPTION
**Pull request by anna1901**

### CODE REVIEW

**Review by pawelgil — COMMENTED** [review]
Such bool-flag approach is a bit troublesome, it causes a tech debit and should be avoided:
* It modifies existing core code (namely the controller, viewmodel etc. that are basically doing their jobs of calculating stats just fine as they are now).
* Adds complexity - the more bool permutations we add the messier the code gets. We can already see how it turns out in some of our views with massive if/else if/else if/... blocks.
* Closes viewmodel and controller for easy extension - before the change those were happily classes unaware of measurement types and all of our app logic. This module basically did its job of computing low/high/avg and providing info about how to display this data. It could without a change be used in another, completely different project (for example a racing game where we'd show lap times in it - high/low/avg? - Note that this is not the goal of open/closed, we won't ever use this in another project for real, but this is a good mental practice to think about modules this way because it validates the architecture). Now its tied to AirCasting and temperature. This is important because HabitatMap is really an open platform and it's really possible that some day we'll be asked to support custom streams with unknown types. This is already possible on the platform level.

I'll put up a PR to this branch with a polymorphic approach to this problem that will not touch current code at all and will still get the job done.

File: AirCasting/Graph/StatisticsContainer/Adapters/ConveringStatisticsDataSourceDecorator.swift
Comment (pawelgil) [inline AirCasting/Graph/StatisticsContainer/Adapters/ConveringStatisticsDataSourceDecorator.swift:15]: plz add `setupHooks()` here

File: AirCasting/SessionViews/ChartViewModel.swift
Comment (pawelgil) [inline AirCasting/SessionViews/ChartViewModel.swift:38]: not a fan of passing a flag here too, but this would require a solid rewrite so maybe let's pass.

But... I think we'll have to start doing such refactorings along the way soon in order to clean the code up a bit. Let's wait for injection mechanism to be merged in first tho.

**Review by pawelgil — CHANGES_REQUESTED** [review]

**Review by pawelgil — APPROVED** [review]

**Review by jankrzempek — APPROVED** [review]

File: AirCasting/SessionViews/SingleMeasurementViewModel.swift
Comment (pawelgil) [inline AirCasting/SessionViews/SingleMeasurementViewModel.swift:8]: well, this is not a correct approach to create viewmodel and still expose the original dependency thru it. I see 2 options here:
1) make this `UserSettings` dependency property private and expose some proper primitive value. Best would be to move the whole value here:
```Swift
@Published var measurementValue: Double
```
this way we can kill two birds with one stone - move some non-view logic from the `SingleMeasurementView` and hide this dependency.

2) Don't introduce this viewmodel just yet and instead properly implement `ObjervableObject` for `UserSettings` (currently this conformance is there only to be able to inject it via `@EnvironmentObject`, we can make it properly observable by calling `objectWillChange.send()` in each of the setters). This method would be easier to implement for sure, but will introduce some more tech debit instead of cleaning things up.

your call :)

**Review by pawelgil — CHANGES_REQUESTED** [review]

File: AirCasting/SessionViews/SingleMeasurementViewModel.swift
Comment (anna1901) [reply AirCasting/SessionViews/SingleMeasurementViewModel.swift]: Okay, I'm choosing the second one, thanks 👌🏻

File: AirCasting/SessionViews/SingleMeasurementView.swift
Comment (pawelgil) [inline AirCasting/SessionViews/SingleMeasurementView.swift:8]: this should be removed

File: AirCasting/AuthViews/UserSettings.swift
Comment (pawelgil) [inline AirCasting/AuthViews/UserSettings.swift:53]: maybe let's add this to each setter, not only the celcius one ;)

**Review by pawelgil — APPROVED** [review]

**Pull request merged**
